lightning/src/chain/keysinterface.rs

```diff
@@ -412,6 +412,17 @@ pub trait KeysInterface {
 	/// This method must return the same value each time it is called with a given `Recipient`
 	/// parameter.
 	fn get_node_secret(&self, recipient: Recipient) -> Result<SecretKey, ()>;
+	/// Get node id based on the provided [`Recipient`]. This public key corresponds to the secret in
+	/// [`get_node_secret`].
+	///
+	/// This method must return the same value each time it is called with a given `Recipient`
+	/// parameter.
```

Comment:
Could precise an API error is return if the type of `Recipient` isn't supported by the implementation. Good to warn caller.